src/ucp/wireup/select.c

```diff
@@ -872,21 +874,27 @@ static ucs_status_t ucp_wireup_add_am_bw_lanes(ucp_ep_h ep, const ucp_ep_params_
     found         = 0;
     tl_bitmap     = -1;
     remote_md_map = -1;
-    for (;;) {
+    /* 1 lane is always AM lane */
+    for (; found < ep->worker->context->config.ext.max_eager_lanes;) {
         status = ucp_wireup_select_transport(ep, address_list, address_count,
                                              &criteria, tl_bitmap, remote_md_map,
                                              !found, &rsc_index, &addr_index, &score);
         if (status != UCS_OK) {
             break;
         }
 
+        remote_cap_flags = address_list[addr_index].iface_attr.cap_flags;
+        is_proxy = !ucp_worker_is_tl_p2p(ep->worker, rsc_index) &&
```

Comment:
can we add common function which checks is_proxy for both am_lane and am_bw_lane , instead of copy/paste?

Reply:
added new function